tests/api/test_adgroup.py

```diff
@@ -0,0 +1,113 @@
+from nose.tools import assert_equal
+from flask import url_for, json
+from tests.base import BaseTestCase
+from tests.populate_database import parse_csv
+from collections import defaultdict
+
+
+class TestAdgroup(BaseTestCase):
```

Comment:
Tests look good. I think a few comments and/or docstrings will make it easier to read them. One thing I think is missing is to check that stuff actually makes it to the database. The API could be lying.